src/content/content.ts

```diff
@@ -2085,6 +2085,23 @@ export class ContentHandler {
         }
         next = ++next % cycleOrder.length;
       }
+    } else if (dictToShow === 'prev') {
+      dict = this.currentDict;
+
+      let prev =
+        (cycleOrder.length + (cycleOrder.indexOf(this.currentDict) - 1)) %
+        cycleOrder.length;
+      while (cycleOrder[prev] !== this.currentDict) {
+        const prevDict = cycleOrder[prev];
+        if (
+          this.currentSearchResult[prevDict] ||
+          (prevDict === 'words' && !!this.currentLookupParams?.meta)
+        ) {
+          dict = prevDict;
+          break;
+        }
+        prev = --prev % cycleOrder.length;
```

Comment:
I think this isn't quite right. In the very unlikely case that we have content in the 'kanji' and 'names' tab but not the word tab, when we swipe left from the 'kanji' tab `prev` will be 1, then in line 2091 it will go to 0. Here we'll calculate `-1 % 3` which gives -1. That's because JS's `%` operator is a remainder operator, not a true `mod` operator.

We actually have a proper `mod()` function in `src/utils/mod.ts` for just this sort of case (we use it in `src/content/content.ts` for this). So I think we want this to be:

```suggestion
        prev = mod(--prev, cycleOrder.length);
```

Reply:
Argh, you're right. Good catch.